### PR TITLE
Initial typescript

### PR DESCRIPTION
**Pull request by Linux123123**
I have added typescript, split up 1 file into multiple files, made all the triggers load up to a Discord Collection for faster search and response, added better logging, ability to change prefix, updated many packages, removed 0.7 triggers and made OCR (hopefully) work better. Also made that if text includes more then 1 trigger they would all be added to message. And lastly, I have replaced normal messages with Embeds!

This is not a drop in replacement, because the API token in stored in a different location. The `npm i --production` will ask for the TOKEN and PREFIX.

### CODE REVIEW

**Comment by google-meet-meat** [comment]
Great!

**Review by Sam1370 — APPROVED** [review]
Great!

**Comment by Sam1370** [comment]
One thing though, many people still prefer 0.7, I don't think there's a need to remove this backwards compatibility

**Comment by vilhelmprytz** [comment]
Is `dist` supposed to be included in versioning?

**Comment by Linux123123** [comment]
> One thing though, many people still prefer - or just prefer not to take the time to update to - 0.7, I don't think there's a need to remove this backwards compatibility

Yeah I guess I can add those commands back, just that we don't really support 0.7 in the server anymore.

> Is `dist` supposed to be included in versioning?

Yes, because 
1. The `--production` flag for `npm` only installs non dev dependencies. (so no `typescript` to build)
2. Not needing to recompile before running is much more simple, then needing to compile. (errors and other things)
3. The `postinstall` which runs after dependency installation and asks for token and prefix, would harded to implement.

**Comment by vilhelmprytz** [comment]
I don't know anything specific to TypeScript, but generally it's bad practice to commit and version compiled files. I don't think it should be included in the repo, if possible.

**Comment by Linux123123** [comment]
> I don't know anything specific to TypeScript, but generally it's bad practice to commit and version compiled files. I don't think it should be included in the repo, if possible.

Ok, no problem. I have pushed out the commit now.
https://github.com/Sam1370/pis-bot/pull/6/commits/90faae486677d6457e171bce2a1c67393dc9f845

**Comment by Linux123123** [comment]
Everything should work now. It should be ready for merging. Just waiting for @vilhelmprytz approval.

**Comment by Sam1370** [comment]
Cool, very nice work!

File: README.md
Comment (vilhelmprytz) [inline README.md:6]:
```suggestion
This is a support bot for the [Pterodactyl Installation Script Discord](https://pterodactyl-installer.se/discord), but feel free to adapt it for your own uses. It is a simple bot that responds to commands, keywords or parsed image text.
```

File: config.js.example
Comment (vilhelmprytz) [inline config.js.example:1]: Shouldn’t this file be TypeScript?

Comment (Linux123123) [reply config.js.example:1]: So, this file isn't even used anymore so will be deleted, but to answer the question, no it shouldn't be typescript. When I converted from compiled in repo to user compilation, I saw that the `setup.ts` file would not be compiled to use before `config.ts` is compiled. So what it did before (and kinda similarly now) is that it would fill in the file from `config.js.example` and write it to the `dist/config/config.js` file. Now it gets the compiled `config.js` replaces it's content with correct details and gets `config.ts` and also replaces contents with correct details (basically replace the details in compiled version to not compile again). So on the next compile (or update) it will automatically compile with correct content.

Comment (vilhelmprytz) [reply config.js.example]: It seems like a complicated way to do this, is there any other way that is less complicated? :slightly_smiling_face:

**Comment by vilhelmprytz** [comment]
Nice work! Is the project supposed to run inside Pterodactyl now (I saw the egg files)? Would it be possible to add a Dockerfile so I can run it using docker-compose (just a normal Docker image)?

**Comment by vilhelmprytz** [comment]
Sorry, I just did some very annoying conflicts I realize. My plan is to merge #7. Once it's merged, can you resolve the conflicts and whatnot? The GitHub actions must be updated to work with TypeScript.

**Comment by Linux123123** [comment]
> Sorry, I just did some very annoying conflicts I realize. My plan is to merge #7. Once it's merged, can you resolve the conflicts and whatnot? The GitHub actions must be updated to work with TypeScript.

The conflicts are now resolved and merged, the GitHub workflow issues are also resolved now and the build passes without errors/warnings.

File: README.md
Comment (vilhelmprytz) [inline README.md:25]:
```suggestion
git clone https://github.com/pterodactyl-installer/pis-bot.git
```

**Review by vilhelmprytz — COMMENTED** [review]
Are the installation instructions in the README of the PR up-to-date? I'd like to try the bot locally so I would need to have instructions on how to get it running both with/without Docker and docker-compose (without Docker just for local testing) now that it's using TypeScript. It looks to me like you've changed quite a few things in the README (perhaps it's already up-to-date) but I just wanted to make sure!

File: README.md
Comment (Linux123123) [reply README.md]: Just updated 👍🏿

**Review by vilhelmprytz — APPROVED** [review]
I still have no idea what I'm doing. But hey, it runs fine locally so LGTM!

**Pull request merged**
